src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -151,64 +127,43 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
         }
     }
 
-    private void getResourcesAndExecute(
-        String workflowId,
-        List<ProcessNode> provisionProcessSequence,
-        ActionListener<WorkflowResponse> listener
-    ) {
-        GetWorkflowStateRequest getStateRequest = new GetWorkflowStateRequest(workflowId, true);
-        client.execute(GetWorkflowStateAction.INSTANCE, getStateRequest, ActionListener.wrap(response -> {
-            // Get a map of step id to created resources
-            final Map<String, ResourceCreated> resourceMap = response.getWorkflowState()
-                .resourcesCreated()
-                .stream()
-                .collect(Collectors.toMap(ResourceCreated::workflowStepId, Function.identity()));
-
-            // Now finally do the deprovision
-            executeDeprovisionSequence(workflowId, resourceMap, provisionProcessSequence, listener);
-        }, exception -> {
-            String message = "Failed to get workflow state for workflow " + workflowId;
-            logger.error(message, exception);
-            listener.onFailure(new FlowFrameworkException(message, ExceptionsHelper.status(exception)));
-        }));
-    }
-
     private void executeDeprovisionSequence(
         String workflowId,
         Map<String, ResourceCreated> resourceMap,
-        List<ProcessNode> provisionProcessSequence,
         ActionListener<WorkflowResponse> listener
     ) {
+
         // Create a list of ProcessNodes with the corresponding deprovision workflow steps
-        List<ProcessNode> deprovisionProcessSequence = provisionProcessSequence.stream()
-            // Only include nodes that created a resource
-            .filter(pn -> resourceMap.containsKey(pn.id()))
-            // Create a new ProcessNode with a deprovision step
-            .map(pn -> {
-                String stepName = pn.workflowStep().getName();
-                String deprovisionStep = getDeprovisionStepByWorkflowStep(stepName);
-                // Unimplemented steps presently return null, so skip
-                if (deprovisionStep == null) {
-                    return null;
-                }
-                // New ID is old ID with deprovision added
-                String deprovisionStepId = pn.id() + DEPROVISION_SUFFIX;
-                return new ProcessNode(
+        List<ProcessNode> deprovisionProcessSequence = new ArrayList<>();
+        for (Map.Entry<String, ResourceCreated> resource : resourceMap.entrySet()) {
```

Comment:
The entryset's order with this map is nondeterministic.  The order of deprovisioning is somewhat robust to failed attempts with retries, but we can reduce those retries by deprovisioning in the reverse order we provisioned (see `Collections.reverse()` at line 168 which reversed the process node sequence).

However we do know the order of entries in the resources created list is the creation order so we can preserve that. We just need to make sure we preserve map insertion order (see next comment where we create the map).